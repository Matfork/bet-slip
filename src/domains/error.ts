export interface ErrorResponse {
  status: number;
  message: string;
}

export class ApiError extends Error {
  status: number;

  constructor({ message, status }: { message: string; status: number }) {
    super(message); // (1)
    this.status = status; // (2)
  }
}
