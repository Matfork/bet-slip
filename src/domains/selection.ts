export enum SelectedSelectionOptions {
  Select,
  Unselect,
}

export interface SelectionType {
  id: string;
  name: string;
  price: number;
}
