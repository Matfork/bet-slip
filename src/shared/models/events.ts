export interface EventType {
  id: string;
  name: string;
  markets: MarketType[];
}

interface MarketType {
  id: string;
  name: string;
  markets: SelectionType[];
}

interface SelectionType {
  id: string;
  name: string;
  price: number;
}
