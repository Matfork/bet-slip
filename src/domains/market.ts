import { SelectionType } from './selection';

export interface MarketType {
  id: string;
  name: string;
  selections: SelectionType[];
}
