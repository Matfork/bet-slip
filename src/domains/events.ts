import { MarketType } from './market';

export interface EventType {
  id: string;
  name: string;
  markets: MarketType[];
}
