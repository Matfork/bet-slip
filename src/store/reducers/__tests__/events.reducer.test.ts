import { SelectedSelectionOptions } from 'domains/selection';
import { EventsActionCreators } from '../../actions/action-creators/events.actionCreator';
import eventsReducer, { initialState } from '../events.reducer';

describe('Events Reducer', () => {
  it('should match snapshot', () => {
    const state = eventsReducer(initialState, {
      type: 'other',
    } as any);

    expect(state).toMatchSnapshot();
  });

  it('should perform SetEvents action', () => {
    const state = eventsReducer(initialState, {
      type: EventsActionCreators.SetEvents,
      payload: [],
    });
    expect(state).toMatchSnapshot();
  });

  it('should perform SerUserSelection action', () => {
    let state = eventsReducer(initialState, {
      type: EventsActionCreators.SetUserSelection,
      payload: {
        option: SelectedSelectionOptions.Select,
        selection: {
          id: '1',
          name: 'test',
          price: 2400,
        },
      },
    });
    expect(state).toMatchSnapshot();

    state = eventsReducer(initialState, {
      type: EventsActionCreators.SetUserSelection,
      payload: {
        option: SelectedSelectionOptions.Unselect,
        selection: {
          id: '1',
          name: 'test',
          price: 2400,
        },
      },
    });
    expect(state).toMatchSnapshot();
  });
});
