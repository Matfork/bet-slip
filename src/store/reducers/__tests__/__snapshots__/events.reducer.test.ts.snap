// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Reducer should match snapshot 1`] = `
Object {
  "events": Array [],
  "userSelection": Array [],
}
`;

exports[`Events Reducer should perform SerUserSelection action 1`] = `
Object {
  "events": Array [],
  "userSelection": Array [
    Object {
      "id": "1",
      "name": "test",
      "price": 2400,
    },
  ],
}
`;

exports[`Events Reducer should perform SerUserSelection action 2`] = `
Object {
  "events": Array [],
  "userSelection": Array [],
}
`;

exports[`Events Reducer should perform SetEvents action 1`] = `
Object {
  "events": Array [],
  "userSelection": Array [],
}
`;
